File: Models/Result.cs
using TravelPickerApp.Models;

namespace TravelPickerApp.Models;

public class Result<T>
{
    public T Data { get; set; }
    public string Message { get; set; }
    public ActionStatusCode Code {  get; set; }
    public Result(ActionStatusCode code,T data,string message)
    {
        this.Code = code;
        this.Data = data;
        this.Message = message;

    }
}

File: ClientApp/Public/src/models/Result.ts
import { IApiResult } from "../types";
import { ActionStatusCode } from "./AppCodes";

export default class ApiResult<T> implements IApiResult<T> {
    public code: number;
    public data: T;
    public message: string;
    constructor(code: number, data: T, message: string) {
        this.data  = data
        this.message = message
        this.code = code
    }
    static ConstructUnexpectedError(){
        return new ApiResult(ActionStatusCode.UnexpectedError,"","Unexpected error occured")
    }
}